### PR TITLE
CA-166738: Fix iSCSI exceptional issue of target without IP

### PR DESCRIPTION
**Pull request by DeliZhangX**
Signed-off-by: Deli Zhang <deli.zhang@citrix.com>

### CODE REVIEW

**Comment by DeliZhangX** [comment]
XenCert iSCSI Job 2764132 passed

**Comment by DeliZhangX** [comment]
Thanks @minli1 @xihuan-citrix

File: src/XenCert/StorageHandler.py [hunk 1]
```diff
@@ -2165,8 +2165,8 @@ def Create(self, device_config = {}):
     def GetPathStatus(self, device_config):
         # Query DM-multipath status, reporting a) Path checker b) Path Priority handler c) Number of paths d) distribution of active vs passive paths
         try:
-            self.mapIPToHost = StorageHandlerUtil._init_adapters()      
-            XenCertPrint("The IP to host id map is: %s" % self.mapIPToHost) 
+            self.mapHostToIP = StorageHandlerUtil._init_adapters()      
+            XenCertPrint("The IP to host id map is: %s" % self.mapHostToIP)
```
Comment (Zhengchai) [inline src/XenCert/StorageHandler.py]: The Printed message mis-matched the parameter name.
The parameter name says "Map Host to IP", the printed message says "The IP to Host ID Map".
which is confusing indeed. can we make it correct?

File: src/XenCert/StorageHandlerUtil.py [hunk 2]
```diff
@@ -73,29 +73,20 @@
...
-                port = util.get_single_entry(glob.glob(\
-                   '/sys/class/iscsi_host/host%s/device/session*/connection*/iscsi_connection*/persistent_port' % host)[0])
-
-            entry = "%s:%s" % (addr,port)
-            adapter[entry] = host
+                addr = util.get_single_entry(glob.glob(
+                    '/sys/class/iscsi_host/host%s/device/session*/connection*/iscsi_connection*/persistent_address' % host)[0])
+                port = util.get_single_entry(glob.glob(
+                    '/sys/class/iscsi_host/host%s/device/session*/connection*/iscsi_connection*/persistent_port' % host)[0])
+            adapter[host] = (addr, port)
```
Comment (Zhengchai) [inline src/XenCert/StorageHandlerUtil.py]: If this line is not aligned with above (i.e. out of the else branch scope), please add a blank line above it.

should this line be included in the else statement branch?

File: src/XenCert/StorageHandlerUtil.py [hunk 3]
```diff
@@ -73,29 +73,20 @@
 def _init_adapters():
     # Generate a list of active adapters
     ids = scsiutil._genHostList(ISCSI_PROCNAME)
-    util.SMlog(ids)
+    util.SMlog("Host ids: %s" % ids)
     adapter = {}
     for host in ids:
         try:
-            # For Backward compatibility 
-            if hasattr(iscsilib, 'get_targetIQN'):
```
Comment (Zhengchai) [inline src/XenCert/StorageHandlerUtil.py]: Why do we remove this targetIQN check? 
It make sense we check here IMO.

**Pull request merged**